### PR TITLE
Refactor to classes

### PR DESCRIPTION
**Pull request by AlierPM**
* In this project, we refactored the Awesome books application to use JavaScript classes.
* [Live Demo](https://walterokumu.github.io/awesome-books/)

### CODE REVIEW

File: scripts/index.js [hunk 1]
```diff
@@ -1,132 +1,132 @@
...
-      Author: author.value,
-    });
+    if (bookTitle.value === '' || author.value === '') {
+      // eslint-disable-next-line no-alert
+      alert('Please make sure to fill both Title & Author fields!');
+    } else {
+      this.bookList.push({
+        Title: bookTitle.value,
+        Author: author.value,
+      });
```
Comment (thecodechaser) [inline scripts/index.js:105]: - [ ] Adding a book to the collection is not working, When I'm trying to add a book it shows this error on the console. Please fix this issue and make sure to maintain all of the original functionality of the application.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @WalterOkumu, and @AlierPM👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: scripts/index.js (hunk 1 above)
Comment (WalterOkumu) [reply scripts/index.js:105]: Hi Ranjeet, I'm unable to replicate your issue. Could you walk me through what exactly has caused the error? On my end I'm able to add and remove with no console errors

Comment (thecodechaser) [reply scripts/index.js:105]: @WalterOkumu I think It's because of the localstorage. The localstorage object wasn't available on my browser.

Comment (WalterOkumu) [reply scripts/index.js:105]: Found it, it was the constructor, bookList wasn't initialised to be an array

Comment (thecodechaser) [reply scripts/index.js:105]: @WalterOkumu good job 👍

**Review by AnselemOdims — APPROVED** [review]
### Hi  @team 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**